Sources/Common/Logger/TidalLogger.swift

```diff
@@ -28,4 +28,30 @@ public extension TidalLogger {
 			line: line
 		)
 	}
+	
+	func log(
+		message: String,
+		source: String,
+		level: Logger.Level? = nil,
+		metadata: [String: String]? = nil,
```

Comment:
[question] shall we consider keeping `Logger.Metadata` type here? So even for the simple cases, people can add other types apart from `String`?

Reply:
Well, I thought to create 2 methods: with `Logger.Metadata` and with `[String: String]` (which will be the majority of the cases in my opinion), but then had problems that Swift couldn't select which function to use 🙂 

I will try to come up with something on Monday, if not - I'll just use `Logger.Metadata` 🙂

Reply:
I think this works fine! 🙌 Once we have more usages also, we can adapt and make appropriate changes if this is not preferred by users of the logging.